### ISSUE TITLE
Content Creator: File upload Input Plugin

### ISSUE THREAD

**Issue by Snugug** [punchcard-cms/input-plugin-file#2]
@scottnath commented on [Thu Aug 04 2016](https://github.com/punchcard-cms/punchcard/issues/195)

We need file uploads. The core idea is the same for both images and files.

``` gherkin
Feature: File uploads
  As a content creator
  I want to upload files during content entry
  So that I may have files as content.

  Scenario: File selection 
    Given that my content type requires me to add a file as data
     When I am entering content in said type
     Then I need to be able to include a file as content
      And I should be able to select a file to upload from my device
      And I should know what file I have selected because it's name is visible after selection.

  Scenario: File types
    Given that my content type requires me to add a file as data
      And that the requirement may be limited to specific file types
     When I am entering content in said type
     Then I should be limited to only allow selection of the file type or types specified by said type.
      And I should not be able to select files which are not that type or types

  Scenario: Upload one file
    Given that my content type limits me to adding only one file as data
     When I am filling in content with a limit to only one file
     Then the file upload input should limit selection to only one file

  Scenario: Validation
    Given that my content type requires me to add a file as data
      And that the requirement is limited to text files
     When I submit a form containing files I'm trying to upload
     Then the validation should check what I have submitted for file uploads
      And it should return an error if any file is not a text file
```

---
- [ ] File upload / File upload
- [ ] File upload / File Types [accept attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-accept)
- [ ] File upload / One
- [ ] File upload / Validation

---

@scottnath commented on [Wed Sep 28 2016](https://github.com/punchcard-cms/punchcard/issues/195#issuecomment-250289386)

blocked by research story: #460 
should build upon #155 

---

@scottnath commented on [Tue Oct 11 2016](https://github.com/punchcard-cms/punchcard/issues/195#issuecomment-253011560)

research story complete. Un-blocking

---

@Snugug commented on [Mon Oct 31 2016](https://github.com/punchcard-cms/punchcard/issues/195#issuecomment-257314977)

Would like a file input plugin in order to work through #492 so tackling this first

**Comment by Snugug** [comment]
@scottnath Looking in to this story, it becomes significantly more difficult with allowing multiple files to be uploaded at once. Can I split that out in to a separate story and rely on repeatable plugins for the time being?

**Comment by scottnath** [comment]
sure!

**Comment by Snugug** [comment]
Split out multiple to https://github.com/punchcard-cms/input-plugin-file/issues/3
